### ISSUE TITLE
Docs - site is confused about current version

### ISSUE THREAD

**Issue by runofthemill** [argoproj/argo-cd#18407]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

This may be an RTD bug/change, but the docs are showing this warning in the header: `You are viewing the docs for an unreleased version of Argo CD`

Screenshot shows the version selected is older, and correctly shows a modal indicating that, but regardless of the selected version, the top banner is present.

Also notable is the window object `READTHEDOCS_DATA` which is used in the [version check](https://github.com/argoproj/argo-cd/blob/1432f11bd99c874ca25558ac1d0e699a30e1c9e7/docs/assets/versions.js#L27) is undefined, making me think it may be a change from RTD.

**To Reproduce**

Visit https://argo-cd.readthedocs.io/en/stable or select any other version

**Expected behavior**

Only show `You are viewing the docs for an unreleased version of Argo CD` when viewing https://argo-cd.readthedocs.io/en/latest/.

**Screenshots**

<img width="1728" alt="SCR-20240524-ijsl" src="https://github.com/argoproj/argo-cd/assets/8418440/4de58948-b158-4f47-97d3-53bf51ce3f6c">

**Version**

N/A

**Logs**

N/A

**Comment by surajyadav1108** [comment]
related #18239

**Comment by morremeyer** [comment]
I built the docs locally with mkdocs directly to investigate

```sh
python3 -m venv /tmp/argocd
source /tmp/argocd/bin/activate
pip install -r docs/requirements.txt

# Install mkdocs if you don't have it, on macOS: brew install mkdocs
mkdocs serve
```

The version selector was fixed in #18239, but the banner still exists when on stable. It looks like selecting `stable` does not actually load the stable docs, since the URL is `/stable`, but the version selector shows `latest`.

Also, I noticed that the 2.10 and 2.11 releases are completely missing from the version selection.

**Comment by andrii-korotkov-verkada** [comment]
I think this is fixed, don't see a banner on stable.
